### PR TITLE
[Model][Boogu-Image] In-repo AutoencoderKL with fused GroupNorm+SiLU

### PR DESCRIPTION
**Pull request by Holworth**
## Summary

This PR adds a model-specific VAE module (`BooguAutoencoderKL`) for Boogu-Image and
replaces every GroupNorm + SiLU chain in its decoder with `fused_group_norm_silu`, to accelerate
the `vae.decode` stage of the pipeline.

Originally, Boogu-Image's VAE came from the third-party `diffusers` library, which makes
it hard to swap its operators for the fused kernel from within the vllm-omni repo. We
therefore rewrite the VAE module in-repo for Boogu-Image (following the same approach as
#6306). The in-repo code is almost a verbatim copy of the third-party implementation and
passes the numerical checks below. A fallback path to the original diffusers VAE module
is also kept. 

## Design & Implementation

The module mirrors the `diffusers` implementation; the design comes down to three pieces: 

- **Module composition.** `BooguResnetBlock2D` / `BooguDecoder` / `BooguEncoder`
  reproduce the VAE's module structure and override `forward()`; the parameterised
  leaves stay stock `nn.GroupNorm` / `nn.Conv2d`, and the mid block reuses diffusers'
  own `Attention`. The unfused paths are **bit-identical** to diffusers.
  `fused_group_norm_silu()` is called only at the decoder's GroupNorm→SiLU sites —
  `norm1`/`norm2` inside `BooguResnetBlock2D` and the decoder's `conv_norm_out`
  (29 sites in total); the encoder is left unfused.
- **Checkpoint loading.** Submodule names match diffusers' exactly
  (`decoder.up_blocks[i].resnets[j]`, `mid_block.attentions[0]`, `conv_norm_out`, …),
  so the stock Boogu `vae/` checkpoint loads with **zero key remap**. `from_pretrained`
  is strict — it raises on any missing/unexpected/mismatched key — and the pipeline
  then falls back to diffusers' `AutoencoderKL` (correct, unfused) with a logged error.
- **Kill-switch.** On by default; disable with
  `additional_config {"vae_group_norm_silu_fusion": false}` or
  `VLLM_OMNI_VAE_GN_SILU_FUSION=0`. Fusion arms only when `config.act_fn == "silu"`,
  and the op falls back to native `F.silu(F.group_norm(...))` where Triton is
  unavailable.

## Correctness

Env: single H200, bf16; Boogu-Image-0.1-Base `vae/` rev `334ad7e5`;
diffusers 0.40.0, torch 2.13.0+cu132, vllm 0.28.0, triton 3.7.1.

- **Weight completeness:** 244/244 keys both directions, zero missing/unexpected;
  every loaded tensor bit-identical to diffusers'.
- **Unfused decode == diffusers, bit-for-bit** at 512² and 1024² (`torch.equal`);
  **encode** mean/logvar bit-equal. The copy is exact.
- **End-to-end image** (seed 42, 1024²×28×cfg4): SSIM 0.9989, PSNR 57.7 dB,
  max |Δ| 3/255 vs the eager baseline.
- Unit suite: **463 passed / 1 skipped** (22 new tests: 19 covering diffusers
  equivalence — every comparison `torch.equal` on the op's native fallback — plus
  weight-load completeness, fusion on/off, site counts and the kill-switch; and 3 for
  strict load / fallback).

## Performance

Same env as above; 10 requests, 2 discarded. `vae.decode` GroupNorm+SiLU: ~29 ms per
1024² request saved (2.0–2.6× per site). End-to-end A/B, fusion ON vs the diffusers
baseline:

- 28-step CFG serving: **−34.9 ms (−0.43%)** engine time.
- 4-step CFG-off (Turbo-style) proxy: **−29.0 ms (−4.22%)**.

`vae.decode` is per-request and step-invariant, so the saving's share scales inversely
with steps — ~0.4% at 28 steps and ~4.2% at the 4-step Turbo proxy, as measured above.
Fusion OFF vs baseline is ≈0 within session noise.

## Test plan

`pytest tests/diffusion/models/boogu_image/ -k "boogu or vae"` (CPU, native fallback,
bit-exact vs diffusers). H200 real-weight parity + A/B as reproduced above.

Part of #6686 / #6665 Track 3 item 1.

### CODE REVIEW

**Comment by hsliuustc0106** [comment]
This PR touches **tests/diffusion/, vllm_omni/diffusion/** (4 files). Based on CODEOWNERS coverage of the changed files, the most-related reviewers appear to be:

> @Bounty-hunter @fhfuih @wtomin

Could one of you take a look when you get a chance? Thanks!

File: vllm_omni/diffusion/models/boogu_image/vae.py [hunk 1]
```diff
@@ -0,0 +1,553 @@
...
+    if isinstance(value, str):
+        return value.strip().lower() not in ("0", "false", "no", "off")
+    return bool(value)
+
+
+def vae_group_norm_silu_fusion_enabled(od_config: Any) -> bool:
+    """Resolve the kill-switch: on by default; ``additional_config
+    {"vae_group_norm_silu_fusion": false}`` or ``VLLM_OMNI_VAE_GN_SILU_FUSION=0``
+    disables it."""
+    if not _as_bool(os.environ.get(ENV_KEY), True):
```
Comment (Bounty-hunter) [inline vllm_omni/diffusion/models/boogu_image/vae.py]: Avoid using environment variables. Also, could you provide the impact on accuracy and performance? Perhaps we could just make it the default and always use the fused implementation.

**Comment by hsliuustc0106** [comment]
Reviewed `7f022f267ae1311f41b83ccc72fb5626e173b26a`.

I found no concrete correctness defect in the decoder-only fusion or strict-loading/fallback path in this snapshot.

The current pre-commit failure is Ruff modifying files; please apply those changes and rebase. For the Turbo benefit, please benchmark the native Turbo and Edit-Turbo checkpoints: the reported four-step CFG-off Base run is a proxy, so its 4.22% gain should remain labeled that way. Please also document the new fusion switch and the replacement VAE's supported scope.

Validation: static source/test review and available CI evidence; I did not run fork code or reproduce GPU benchmarks.
